tests/unit/steps/test_hypervisor.py

```diff
@@ -157,22 +161,124 @@ def test_hypervisor_azs_grouping():
 
     hypervisor_planner_all = HypervisorUpgradePlanner([app1, app2], list(machines.values()))
 
-    assert dict(hypervisor_planner_all.azs) == exp_azs_all
+    assert dict(hypervisor_planner_all.azs(target)) == exp_azs_all
 
     # passing machine 0 to the HypervisorUpgradePlanner
     exp_azs_0 = AZs()
     exp_azs_0["az0"].app_units["app1"] = [units["app1/0"]]
     exp_azs_0["az0"].app_units["app2"] = [units["app2/0"]]
 
     hypervisor_planner_machine_0 = HypervisorUpgradePlanner([app1, app2], [machines["0"]])
-    assert dict(hypervisor_planner_machine_0.azs) == exp_azs_0
+    assert dict(hypervisor_planner_machine_0.azs(target)) == exp_azs_0
 
     # passing machine 1 to the HypervisorUpgradePlanner
     exp_azs_1 = AZs()
     exp_azs_1["az0"].app_units["app1"] = [units["app1/1"]]
 
     hypervisor_planner_machine_1 = HypervisorUpgradePlanner([app1, app2], [machines["1"]])
-    assert dict(hypervisor_planner_machine_1.azs) == exp_azs_1
+    assert dict(hypervisor_planner_machine_1.azs(target)) == exp_azs_1
+
+
+def test_hypervisor_azs_grouping_units_different_os_release():
```

Comment:
Is this really necessary test? It's too complex testing just simple if condition in the `azs` function.

Reply:
Personally I think it's good to keep this test. Unfortunately is too long to create the conditions but it shows some important behaviors like:
- apps in the same machine with different OpenStack releases (one app already upgraded), the not upgraded one will appear in the AZ
- apps already upgraded doesn't show up
- AZs already upgraded doesn't show up

Reply:
I see, but still for me, it's not good, since it's relying on multiple methods / objects, so it can fail without relevant changes in the `get_azs` function. Resolving this for now.